### DISCUSSION TITLE
Dynamic table cells content

### DISCUSSION THREAD

**Discussion by SFU253** [taiga-family/taiga-ui#6590] [Q&A] [answered]
Hi, i am trying to pass a table cells template to a table component, but the rows are not getting rendered.
In this very basic example with static content, i am passing the columns and the template, which is supposed to render the content.
``` HTML
<app-table
  [columns]="['KEY1', 'KEY2', 'KEY3']"
  [columnTemplate]="test"
></app-table>

<ng-template #test>
  <td *tuiCell="'KEY1'" tuiTd>template1</td>
  <td *tuiCell="'KEY2'" tuiTd>template2</td>
  <td *tuiCell="'KEY3'" tuiTd>template3</td>
</ng-template>
```
The table component receives the template as `PolymorpheusContent` and is trying to render it in the body. The exact same layout directly in the component is working fine, but the template will only render empty rows.
``` HTML
<table tuiTable [columns]="columns" class="table">
  <thead tuiThead>
    <tr tuiThGroup>
      <th *tuiHead="'KEY1'" tuiTh>KEY1</th>
      <th *tuiHead="'KEY2'" tuiTh>KEY2</th>
      <th *tuiHead="'KEY3'" tuiTh>KEY3</th>
    </tr>
  </thead>
  <tbody tuiTbody [data]="data">
    <tr *ngFor="let item of data" tuiTr>
      <ng-container *polymorpheusOutlet="columnTemplate as template">{{
        template
      }}</ng-container>
    </tr>
    <tr *ngFor="let item of data" tuiTr>
      <td *tuiCell="'KEY1'" tuiTd>{{ item.key1 }}</td>
      <td *tuiCell="'KEY2'" tuiTd>{{ item.key2 }}</td>
      <td *tuiCell="'KEY3'" tuiTd>{{ item.key3 }}</td>
    </tr>
  </tbody>
</table>
```
![image](https://github.com/taiga-family/taiga-ui/assets/44519378/bbb769b7-6fd3-4f8f-ad1d-e91ecd7f5801)

Am i doing something terrible wrong, or is this use case not supported by the table? 
Here is a link to the full example on Stackblitz -> https://stackblitz.com/edit/angular-tck61i?file=src%2Fapp%2Fapp.component.html

Every help is appreciated! Thank you very much

**Comment by waterplea** [comment, marked as answer]
That's because directives are connected together with DI, and in your case they are instantiated away from the table itself. I think this might get fixed in 4.0, once polymorpheus starts passing injector to templates. In current version it is impossible due to Angular 12.
